app/code/Magento/Ui/view/base/web/js/grid/columns/image-preview.js

```diff
@@ -130,6 +131,11 @@ define([
         show: function (record) {
             var img;
 
+            if (!this.isListenerActive) {
```

Comment:
I believe we can check for `visibleRecord` in `handleKeyDown` without introducing the new field and show method logic complication